src/Scriban/Functions/ArrayFunctions.cs

```diff
@@ -206,6 +198,41 @@ private static IEnumerable EachInternal(TemplateContext context, SourceSpan span
             }
         }
 
+        /// <summary>
+        /// Filters the input list according the supplied filter function.
+        /// </summary>
+        /// <param name="context">The template context</param>
+        /// <param name="span">The source span</param>
+        /// <param name="list">An input list</param>
+        /// <param name="function">The function used to test each elemement of the list</param>
+        /// <returns>Returns a new list which contains only those elements which match the filter function.</returns>
+        /// <remarks>
+        /// ```scriban-html
+        /// {{func large(x) ; ret x>=100 ; end}}
+        /// {{ [1, 200, 3,400] | array.each @large }}
+        /// ```
+        /// ```html
+        /// [200, 400]
+        /// ```
+        /// </remarks>
+        public static ScriptRange Filter(TemplateContext context, SourceSpan span, IEnumerable list, object function)
+        {
+            return ApplyFunction(context, span, list, function, FilterInternal);
```

Comment:
Can you cache FilterInternal to a static readonly field with directly the delegate. Same for the other calls. That will avoid the allocation of this delegate everytime we call ApplyFunction.

Reply:
I *believe* (but am prepared to be corrected) that the same result can be achieved using s static local function...
      ```
  public static ScriptRange Filter(TemplateContext context, SourceSpan span, IEnumerable list, object function)
        {
            static IEnumerable FilterInternal(TemplateContext context, SourceSpan span, IEnumerable list, IScriptCustomFunction function, Type destType)
            {
                ....
            }
            return ApplyFunction(context, span, list, function, FilterInternal);
        }
```
To my mind this is a nicer syntax since it bundles everything in one place but if you prefer "old" syntax I can use that.

Reply:
Not really, have a look at [this example](https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQMwAJboMLoN7LpGYZRwBsmALOgLIAUAlPocW2ZVDQDID2AxgEMANk3wBfVmyIARXvT5DhjANxT0kpG3VpMFaujn0yMdAFtGeTeKA==) notice the allocation of the delegate

Reply:
Ok - have introduced an explicit delegate type to handle this.